### PR TITLE
Add booleans concept

### PR DESCRIPTION
**Pull request by mfoda**

### CODE REVIEW

**Comment by mfoda** [comment]
> /format

Is this the correct way to invoke the formatter, and where can I see the workflow output?

**Comment by mfoda** [comment]
A bit of feedback on the wording used in `instructions.md` for the `free prisoner` conditions:

> _Free prisoner_: if the prisoner is awake and the other two characters are sleeping, a sneaky entry into the camp can free the prisoner. This won't work if the prisoner is sleeping, as the prisoner will be startled by the sudden appearance of her friend and the knight and archer will be awoken. The prisoner can also be freed if the archer is sleeping and Annalyn has her pet dog with her, as the knight will be scared by the dog and will withdraw, and the archer can't equip his bow fast enough to prevent the prisoner from being freed.

I'm not sure if it's just me but the way this read to me is as follows:
- The prisoner must always be awake to be freed 
- And both the knight and archer must be asleep (free cond 1)
- Or the knight can be awake if the dog is there, but the archer must be asleep (free cond 2)

I was a bit surprised to find the answer to be
```csharp
return !knightIsAwake && !archerIsAwake && prisonerIsAwake || !archerIsAwake && petDogIsPresent;
```
as the wording did not make it sound like it's possible to free a sleeping prisoner under any conditions.

Maybe the following is more clear?

> _Free prisoner_: if the prisoner is awake and the other two characters are sleeping, a sneaky entry into the camp can free the prisoner. This won't work if the prisoner is sleeping, as the prisoner will be startled by the sudden appearance of her friend and the knight and archer will be awoken. However, a sleeping prisoner can be freed if the archer is also sleeping and Annalyn has her pet dog with her, as the knight will be scared by the dog and will withdraw, and the archer can't equip his bow fast enough to prevent the prisoner from being freed.

**Comment by ErikSchierboom** [comment]
> Maybe the following is more clear?

It might be useful to post this in the #v3 channel on Slack, as there are several tracks using the above wording.

**Comment by ErikSchierboom** [comment]
> Is this the correct way to invoke the formatter, and where can I see the workflow output?

The formatter was only enabled for the v3 repo: https://github.com/exercism/v3/blob/main/.github/workflows/format-code.yml You could try and copy that to this repo, but it might be tricky to get working.

File: exercises/concept/annalyns-infiltration/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+- _Spy_: the group can be spied upon if at least one of them is awake. Otherwise, spying is a waste of time.
+- _Signal prisoner_: the prisoner can be signalled using bird sounds if the prisoner is awake and the archer is sleeping, as archers are trained in bird signaling so they could intercept the message.
+- _Free prisoner_: if the prisoner is awake and the other two characters are sleeping, a sneaky entry into the camp can free the prisoner. This won't work if the prisoner is sleeping, as the prisoner will be startled by the sudden appearance of her friend and the knight and archer will be awoken. The prisoner can also be freed if the archer is sleeping and Annalyn has her pet dog with her, as the knight will be scared by the dog and will withdraw, and the archer can't equip his bow fast enough to prevent the prisoner from being freed.
+
+You have four tasks: to implement the logic for determining if the above actions are available based on the state of the three characters found in the forest and whether Annalyn's pet dog is present or not.
+
+## Tasks
```
Comment (ErikSchierboom) [inline exercises/concept/annalyns-infiltration/.docs/instructions.md:20]: Most tracks don't have this additional heading, so perhaps remove it?

```suggestion
```

File: concepts/booleans/about.md [hunk 2]
```diff
@@ -1 +1,23 @@
-TODO about: booleans
\ No newline at end of file
+Booleans in Haxe are represented by the `Bool` type, whose values can be either `true` or `false`.
+
+Haxe supports three [boolean operators][operators]: `!` (unary NOT), `&&` (binary AND), and `||` (binary OR)
+
+Compound boolean expressions (`&&` and `||`) use _short-circuit_ evaluation, where expressions are evaluated from left to right but only as far as necessary. This is significant in cases such as the following:
+
+```haxe
+if (object != null && object.field == x) {}
```
Comment (ErikSchierboom) [inline concepts/booleans/about.md:8]: Looking at the prerequisites of this exercise, it looks like students might not be familiar with `if` statements. As such, maybe rewrite this to:

```suggestion
object != null && object.field == x
```

**Review by ErikSchierboom — APPROVED** [review]
Just some tiny nits

**Comment by mfoda** [comment]
@ErikSchierboom thanks for the suggestions!

**Pull request merged**